### PR TITLE
Provide additional options to choose when rejecting a contribution

### PR DESCRIPTION
**Pull request by gasparRemoteWork**
We added a mutation to unsubmit tasks The claim is reverted to the active state

### CODE REVIEW

File: api/work/mutations.py [hunk 1]
```diff
@@ -915,6 +915,38 @@ def mutate(current_person, info, *args, **kwargs):
             return RejectTaskMutation(success=False, message="The task doesn't exist")
 
 
+class UnSubmitTaskMutation(InfoStatusMutation, graphene.Mutation):
```
Comment (ashikzk) [inline api/work/mutations.py:918]: Please refactor `UnSubmit` with `RequestRevision` in all places.

**Review by ashikzk — COMMENTED** [review]
@gasparRemoteWork Both the front-end and back-end implementation looks good. Only change that we need is a refactor of `UnSubmit` with `RequestRevision`. Once that is done, we can accept this PR.

File: api/work/mutations.py (hunk 1 above)
Comment (gasparRemoteWork) [reply api/work/mutations.py]: Addressed

**Review by ashikzk — COMMENTED** [review]
Apart from the above clarification, everything else looking good to be merged.

File: api/work/mutations.py [hunk 2]
```diff
@@ -915,6 +915,38 @@ def mutate(current_person, info, *args, **kwargs):
...
+        try:
+            task_id = kwargs.get("task_id")
+            task = Task.objects.get(id=task_id)
+
+            # check if user has permissions
+            if not is_admin_or_manager(current_person, task.product.slug):
+                return RequestRevisionTaskMutation(success=False, message="You don't have permissions")
+
+            # set "Failed" status to task claims
+            task_claim = task.taskclaim_set.filter(kind__in=[CLAIM_TYPE_DONE, CLAIM_TYPE_IN_REVIEW]).all()
```
[the remark this answers is not in the document]
Comment (gasparRemoteWork) [reply api/work/mutations.py:934]: I implemented the same logic we do for rejecting a task (screenshot). From what I understand of the task description, requesting a revision is the same as rejecting a task but keeping the user who claimed it.
But if it is not the case, I will change it
image

[the remark this answers is not in the document]
Comment (ashikzk) [reply api/work/mutations.py:934]: Ok, got it. In that case, let's keep it like that. Thanks for the clarification.

**Pull request merged**